CRM/Dedupe/Merger.php

```diff
@@ -624,15 +623,14 @@ public static function moveContactBelongings($mergeHandler, $tables, $tableOpera
    * @param array $cidRefs
    *
    * @throws \CRM_Core_Exception
-   * @throws \Civi\API\Exception\UnauthorizedException
    */
   protected static function filterRowBasedCustomDataFromCustomTables(array &$cidRefs) {
-    $customTables = (array) CustomGroup::get(FALSE)
-      ->setSelect(['table_name'])
-      ->addWhere('is_multiple', '=', 0)
-      ->addWhere('extends', 'IN', array_merge(['Contact'], CRM_Contact_BAO_ContactType::contactTypes()))
-      ->execute()
-      ->indexBy('table_name');
+    $filters = [
+      'is_multiple' => FALSE,
+      'extends' => 'Contact',
```

Comment:
Just double checking @colemanw it will deal with the primary contact types as well in getAll()

Reply:
answer is yes https://github.com/civicrm/civicrm-core/blob/master/CRM/Core/BAO/CustomGroup.php#L88